stackdriver/client.go

```diff
@@ -120,9 +124,15 @@ func (c *Client) getConnection(ctx context.Context) (*grpc.ClientConn, error) {
 	if len(c.url.Port()) > 0 {
 		address = fmt.Sprintf("%s:%s", address, c.url.Port())
 	}
-	conn, err := grpc.DialContext(ctx, address, dopts...)
-	c.conn = conn
-	return conn, err
+	if c.resolver != nil {
+		conn, err := grpc.DialContext(ctx, c.resolver.Scheme()+":"+address, dopts...)
+		c.conn = conn
+		return conn, err
+	} else {
+		conn, err := grpc.DialContext(ctx, "dns:"+address, dopts...)
+		c.conn = conn
+		return conn, err
+	}
```

Comment:
This could be simplified by introducing a `scheme` variable, because that's the only difference between the two branches.